src/core/legate_c.h

```diff
@@ -84,15 +85,33 @@ typedef enum legate_core_type_code_t {
   UINT16_LT     = LEGION_TYPE_UINT16,
   UINT32_LT     = LEGION_TYPE_UINT32,
   UINT64_LT     = LEGION_TYPE_UINT64,
-  HALF_LT       = LEGION_TYPE_FLOAT16,
-  FLOAT_LT      = LEGION_TYPE_FLOAT32,
-  DOUBLE_LT     = LEGION_TYPE_FLOAT64,
+  FLOAT16_LT    = LEGION_TYPE_FLOAT16,
+  FLOAT32_LT    = LEGION_TYPE_FLOAT32,
+  FLOAT64_LT    = LEGION_TYPE_FLOAT64,
   COMPLEX64_LT  = LEGION_TYPE_COMPLEX64,
   COMPLEX128_LT = LEGION_TYPE_COMPLEX128,
-  STRING_LT     = COMPLEX128_LT + 1,
-  MAX_TYPE_NUMBER,
+  // Compound types
+  FIXED_ARRAY_LT,
+  STRUCT_LT,
+  // Variable size types
+  STRING_LT,
+  INVALID_LT = -1,
 } legate_core_type_code_t;
 
+typedef enum legate_core_reduction_op_kind_t {
+  ADD_LT = LEGION_REDOP_KIND_SUM,
+  SUB_LT = LEGION_REDOP_KIND_DIFF,
+  MUL_LT = LEGION_REDOP_KIND_PROD,
+  DIV_LT = LEGION_REDOP_KIND_DIV,
+  MAX_LT = LEGION_REDOP_KIND_MAX,
+  MIN_LT = LEGION_REDOP_KIND_MIN,
+  OR_LT  = LEGION_REDOP_KIND_OR,
+  AND_LT = LEGION_REDOP_KIND_AND,
+  XOR_LT = LEGION_REDOP_KIND_XOR,
+} legate_core_reduction_op_kind_t;
+
+#define MAX_TYPE_NUMBER (COMPLEX128_LT + 1)
```

Comment:
https://github.com/nv-legate/legate.core/pull/697/commits/0a3105b4df9827f5afd2b453c744246635bd7d76#diff-b0130510eeec3744cdd00fb2d06435229c70360346698827f0d4fb3553ebb67aL113

Reply:
Looks like cuNumeric was still using `MAX_TYPE_NUMBER` for no good reason: https://github.com/nv-legate/cunumeric/pull/903/commits/bc46f3c7591920172c9d0a1b213a78f876d4eb64

This actually reminded me of the API change that I was planning but forgot to include in this PR; the reduction operator registration now takes an operator ID as an argument, instead of getting it form a static member: https://github.com/nv-legate/legate.core/pull/697/commits/0a3105b4df9827f5afd2b453c744246635bd7d76#diff-829398d8ad4dbb41f58a5efa025af6c314c5b51cff67bc61bae17c524ecbff48R167